cardano-cli/src/Cardano/CLI/Run.hs

```diff
@@ -54,6 +57,19 @@ data ClientCommand =
   | forall a. Help ParserPrefs (ParserInfo a)
   | DisplayVersion
 
+
+data PreMainnetCommand
+ = PreMainnetTxBuild PreMainnetTransactionBuildCmdArgs
+
+
+-- E.g something like this that offers the minimum required functionality.
+-- We also need key generation commands to work to run SPOs etc
+data PreMainnetTransactionBuildCmdArgs = TransactionBuildCmdArgs
+  { nodeSocketPath          :: !SocketPath
+  , networkId               :: !NetworkId
+  , txins                    :: ![TxIn]
```

Comment:
I didn't fully flesh out this type but you get the idea. We're missing outputs etc.